packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts

```diff
@@ -59,16 +60,28 @@ async function createDataObject<TObj extends PureDataObject, I extends DataObjec
 
     // request mixin in
     runtimeClass = mixinRequestHandler(
-        async (request: IRequest, runtimeArg: FluidDataStoreRuntime) =>
-            (await PureDataObject.getDataObject(runtimeArg)).request(request),
-            runtimeClass);
+        async (request: IRequest, runtimeArg: FluidDataStoreRuntime) => {
+            const maybeRouter: FluidObject<IFluidRouter> & IFluidLoadable | undefined =
+                (await runtimeArg.IFluidLoadable?.handle?.get());
+            assert(maybeRouter?.IFluidRouter !== undefined, "Data store runtime entrypoint is not an IFluidRouter");
+            return maybeRouter?.IFluidRouter.request(request);
+        },
+        runtimeClass);
 
     // Create a new runtime for our data store
     // The runtime is what Fluid uses to create DDS' and route to your data store
-    const runtime = new runtimeClass(
+    const runtime: FluidDataStoreRuntime = new runtimeClass(
         context,
         sharedObjectRegistry,
         existing,
+        async (rt: IFluidDataStoreRuntime) => {
+            assert(instance !== undefined, "Entrypoint is undefined");
+            // Calling finishInitialization here like PureDataObject.getDataObject did, to keep the same behavior,
+            // since accessing the runtime's entrypoint is how we want the data object to be retrieved going forward.
+            // Without this I ran into issues with the load-existing flow not working correctly.
+            await instance.finishInitialization(true);
```

Comment:
should this wrap in a `if(existing)` so it is only called once?

Reply:
i guess this matches the exact behavior as before, which seems odd, but should handle in a follow up

Reply:
https://github.com/microsoft/FluidFramework/pull/11887/files#r984945860

Reply:
In the new version it will only get called once, though, right? Because this ends up being the LazyPromise inside the handle [here](https://github.com/microsoft/FluidFramework/blob/75fc7ff42e58b5d7ac30962830c914b6c4e7f775/packages/runtime/datastore/src/dataStoreRuntime.ts#L313), and it'll only run the first time the handle is accessed.

Reply:
it's called directly on the instance a few lines down. so it will be called twice in the !existing case, and once in all other cases

Reply:
i think the current code is fine, and doesn't change existing behavior, we can always open an issue to make it better